### PR TITLE
chore(deps): upgrade dependencies

### PR DESCRIPTION
**Pull request by angeloashmore**
## Types of changes

- [x] Chore (a non-breaking change which is related to package maintenance)
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

This PR upgrade the package's dependencies. The following notable upgrades are included:

- `@prismicio/types`: Interoperability with `@prismicio/types-internal`.
- `@prismicio/mock`: Supports the mock factory API via `createMockFactory()`.

The PR also adds a `ctx.mock` property to test context for easy mock Prismic model and value generation.

## Checklist:

- [ ] My change requires an update to the official documentation.
- [x] All [TSDoc](https://tsdoc.org) comments are up-to-date and new ones have been added where necessary.
- [x] All new and existing tests are passing.

🐶

### CODE REVIEW

File: vitest.config.ts [hunk 1]
```diff
@@ -5,5 +5,6 @@ export default defineConfig({
 		coverage: {
 			reporter: ["lcovonly", "text"],
 		},
+		setupFiles: ["./vitest.setup"],
```
Comment (lihbr) [inline vitest.config.ts]: I'm wondering if we cannot have the setup file inside of the test folder, just in order to minimize root files, wdyt? À-la AVA perhaps? https://github.com/prismicio/slice-simulator/blob/master/packages/core/test/__testutils__/setup.ts

Comment (angeloashmore) [reply vitest.config.ts]: Works for me! Nesting the setup file in `__testutils__` might make it hard to find. What if we put it in the top level of `test/`, similar to `__fixtures__`, `__testutils__`, and `__snapshots__`?

```
test/__setup__.ts
```

Comment (lihbr) [reply vitest.config.ts]: Works for me! Let's go with it ☺️

**Review by AdrienHallet — APPROVED** [review]
👍

**Pull request merged**